Sources/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -262,6 +262,11 @@ extension ProfileStore {
 	/// Claim the profile by updating **lastUsedOnDevice**
 	func claimProfileSnapshot(_ snapshot: inout ProfileSnapshot) async throws {
 		snapshot.header.lastUsedOnDevice = try await Self.createDeviceInfo()
+		do {
+			try await secureStorageClient.saveDeviceIdentifier(snapshot.header.lastUsedOnDevice.id)
+		} catch {
+			loggerGlobal.critical("Failed to save newly generated device identifier, error: \(error)")
```

Comment:
nothing to do if we fail really... but really should not fail.

Reply:
Can't we retry a few times at least?

Reply:
how many times do you wanna retry? any sleep between tries?

Reply:
2-3 times, sleeping probably doesn't help. But if you are sure it can't fail then we could skip that.